sklearn/ensemble/_forest.py

```diff
@@ -507,7 +508,13 @@ def fit(self, X, y, sample_weight=None):
                     "supported: continuous, continuous-multioutput, binary, "
                     "multiclass, multilabel-indicator."
                 )
-            self._set_oob_score_and_attributes(X, y)
+
+            if inspect.isfunction(self.oob_score):
```

Comment:
Isn't it enough?

```suggestion
            if callable(self.oob_score):
```

Reply:
I'll find out. I had forgotten about the builtin `callable`.

Reply:
I asked the docs because I was interested in what exactly the difference was. `callable(foo) == True` doesn't mean you can actually call `foo`, but `callable(foo) == False` means you can not call it. I think `inspect.function(foo) == True` promises that you can call it, but a method is not a function. 

```
class Foo:
  def score(self):
    pass

isfunction(Foo().score) == False
```

TL;DR: lets use `callable`.